docs/source/teams/data_lens.rst

```diff
@@ -0,0 +1,773 @@
+            self,
+            request: DataLensSearchRequest
+        ) -> Generator[DataLensSearchResponse, None, None]:
+            # Create our client
+            client = bigquery.Client()
+
+            try:
+                # Retrieve our Data Lens search parameters
+                detection_label = request.search_params.get("detection_label", "")
+
+                # Construct our query
+                query = """
+                        SELECT
+                            media_path, tags, detections, keypoints
+                        FROM `my_dataset.samples_json`,
+                        UNNEST(JSON_QUERY_ARRAY(detections)) as detection
+                        WHERE JSON_VALUE(detection.label) = @detection_label
+                    """
+
+                # Submit our query to BigQuery
+                job_config = bigquery.QueryJobConfig(
+                    query_parameters=[
+                        bigquery.ScalarQueryParameter(
+                            "detection_label",
+                            "STRING",
+                            detection_label
+                        )
+                    ]
+                )
+                query_job = client.query(query, job_config=job_config)
+
+                # Wait for results
+                rows = query_job.result(
+                    # BigQuery will handle pagination automatically, but
+                    # we can optimize its behavior by synchronizing with
+                    # the parameters provided by Data Lens
+                    page_size=request.batch_size,
+                    max_results=request.max_results
+                )
+
+                samples = []
+
+                # Iterate over data from BigQuery
+                for row in rows:
+
+                    # Transform sample data from BigQuery format to FiftyOne
+                    samples.append(self.convert_to_sample(row))
+
+                    # Yield next batch when we have enough samples
+                    if len(samples) == request.batch_size:
+                        yield DataLensSearchResponse(
+                            result_count=len(samples),
+                            query_result=samples
+                        )
+
+                        # Reset our batch
+                        samples = []
+
+                # We've run out of rows, but might have a partial batch
+                if len(samples) > 0:
+                    yield DataLensSearchResponse(
+                        result_count=len(samples),
+                        query_result=samples
+                    )
+
+                # Our generator is now exhausted
+
+            finally:
+                # Clean up our client on exit
+                client.close()
+
+Let's take a look at a few parts in detail.
+
+.. code-block:: python
+    :linenos:
+
+    # Create our client
+    client = bigquery.Client()
+
+In practice, you'll likely need to use :ref:`secrets <teams-secrets>` to
```

Comment:
@tom-vx51 note that I added this callout.

In order to support instantiating a client with credentials, shouldn't you move client creation from `BigQueryHandler. handle_request()` into `BigQueryConnector. handle_lens_search_request()` so that you have access to `ctx.secrets`?

I think it would be valuable to clean this up so that readers understand how to inject credentials.

Reply:
yeah I was conflicted on how much to focus on details like this as I didn't want to take the focus away from data lens, but we can certainly make it a bit more realistic

Reply:
ack will include with screenshot updates